File: backend/src/main/java/com/artifax_solutions/website/model/StreamerSection.java
package com.artifax_solutions.website.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class StreamerSection {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ElementCollection
    private List<String> carouselImages;

    private String title;
    private String subtitle;
    private String description;

    @ElementCollection
    private List<String> headings;

    @ElementCollection
    private List<String> texts;
    private String buttonText;
    private String buttonLink;
}

File: frontend/src/interface/StreamerSection.ts
export interface StreamerSection {
  id: number | undefined | null;
  carouselImages?: string[] | undefined | null;
  videoUrl?: string[] | undefined | null | string;
  title: string | undefined | null;
  subTitle: string | undefined | null;
  description: string | undefined | null;
  headings?: string[] | undefined | null;
  texts?: string[] | undefined | null;
  buttonText?: string | undefined | null;
  buttonLink?: string | undefined | null;
}
